pandas/core/ops.py

```diff
@@ -752,31 +758,36 @@ def wrapper(left, right, name=name, na_op=na_op):
         na_op = converted.na_op
 
         if isinstance(rvalues, ABCSeries):
-            name = _maybe_match_name(left, rvalues)
             lvalues = getattr(lvalues, 'values', lvalues)
             rvalues = getattr(rvalues, 'values', rvalues)
             # _Op aligns left and right
         else:
-            if isinstance(rvalues, pd.Index):
-                name = _maybe_match_name(left, rvalues)
-            else:
-                name = left.name
             if (hasattr(lvalues, 'values') and
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
+        res_name = _get_series_op_result_name(left, right)
         result = wrap_results(safe_na_op(lvalues, rvalues))
         return construct_result(
             left,
             result,
             index=left.index,
-            name=name,
+            name=res_name,
             dtype=dtype,
         )
 
     return wrapper
 
 
+def _get_series_op_result_name(left, right):
+    # `left` is always a Series object
+    if isinstance(right, (pd.Series, pd.Index)):
```

Comment:
use generic instance checks, thought why can't you just duck type, e.g.

```
name = _maybe_match_name(left, getattr(right, 'name', None))
```

Reply:
On the generics, sure.  On the ducktype:

1) pretty sure what you have in mind is `name = _maybe_match_name(left, right)`

2) That would represent a non-trivial change in pandas convention/policy for name propagation.  Up until a few days ago even `Index` names were ignored (except for unintentional corner cases caused by conversion within `_TimeOp`).  Allowing through anything with a `name` attribute would include e.g. most `DateOffset` subclasses, which I don't think is desired.  If this is something you'd like to see changed, I'd ask you to open an issue or something and consider it out of scope for this bug-fixing PR.